src/sinks/prometheus/remote_write.rs

```diff
@@ -80,15 +93,34 @@ impl SinkConfig for RemoteWriteConfig {
             buckets,
             quantiles,
         };
-        let sink = request
-            .batch_sink(
-                HttpRetryLogic,
-                service,
-                MetricBuffer::new(batch.size),
-                batch.timeout,
-                cx.acker(),
-            )
-            .sink_map_err(|error| error!(message = "Prometheus remote_write sink error.", %error));
+
+        let sink = {
+            let service = request.service(HttpRetryLogic, service);
+            let service = ServiceBuilder::new().service(service);
+            let buffer = PartitionBuffer::new(MetricBuffer::new(batch.size));
+            PartitionBatchSink::new(service, buffer, batch.timeout, cx.acker())
+                .with_flat_map(move |event: Event| {
+                    let tenant_id = tenant_id.as_ref().and_then(|template| {
+                        template
+                            .render_string(&event)
+                            .or_else(|fields| {
+                                error!(
+                                    message = "Tenant template contains missing fields",
+                                    ?fields,
+                                    rate_limit_secs = 30,
+                                );
```

Comment:
I wasn't sure if this should be an internal event or not, as it can only be caused (currently) by misconfiguration and not by the source events.

Reply:
It's conceivable that a field coming from say the syslog source may intermittently not include a field, the data may occassionally not include the structured data from the field. I'm not sure how likely that would be used here, but I think it suggests it's not necessarily just misconfiguration.

So unless I'm misunderstanding, I'd vote for this to be an internal event.

Reply:
I was thinking of the current situation where the template does not actually make use of the metric data (ref #5985). However, that will likely be resolved in the not too distant future, so this makes sense.

(Note that the syslog source does not output metrics, so is not directly applicable, but the point still stands here)